src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -4097,12 +4105,14 @@ void AdaptDispatch::_ReportSGRSetting() const
     addAttribute(L";1"sv, attr.IsIntense());
     addAttribute(L";2"sv, attr.IsFaint());
     addAttribute(L";3"sv, attr.IsItalic());
-    addAttribute(L";4"sv, attr.IsUnderlined());
+
+    const auto ulStyle = WI_EnumValue(attr.GetUnderlineStyle());
+    addAttribute(fmt::format(FMT_COMPILE(";4:{}"sv), ulStyle), attr.IsUnderlined());
+
```

Comment:
This needs to differentiate between the extended styles and standard styles similar to the way the vt render engine works, i.e. it should be reporting `4` and `21` for the single and double underlines styles, and only using the `4:x` syntax for the extended styles. Applications that limit themselves to ANSI standard sequences will not be expecting colons in the response.

Reply:
I thought we are good with DECRQSS sending colons, like we did with `Index256` and `IsRgb`. They are reported with colon regardless of which type of client requested it. But I'm happy doing it either way.

Reply:
The difference is that the ITU colors are intended to be used with colons, and any app that has used an ITU color should be prepared to receive a colon response, regardless of how it was set. But an app that has restricted itself to ANSI colors should only ever receive an ANSI-compatible response. 

And the same applies to the rendition attributes. If an app has used extended underline styles, then its fine to report those with colon sequences. But if it has only used ANSI attributes then our responses should be ANSI-compatible as well.

Reply:
singly underlined and doubly underlined are now reported with ANSI compatible sequences in _DECRQSS_. Resolving.